### PR TITLE
Cleanup service controller unit test

### LINKED ISSUE #116260: test cleanup needed: cloud-provider/controllers/service/controller_test.go
Similar to #114511

Table-driven tests are awesome, but they can be very verbose and obscure what's actually being tested in each case.

I started on cleaning up staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go, but I only did two tests - there are more.  This is a good opportunity for someone to read some code, so some cleanup, and score a PR or two towards org membership or similar :)

https://github.com/kubernetes/kubernetes/pull/116259

I did: 

https://github.com/kubernetes/kubernetes/commit/3d63aed2dca23bfec17c174b32b63c0be511fe85

after that, it was obvious we have redundant test cases:

https://github.com/kubernetes/kubernetes/commit/b5e4bbb132d0d6c4079e5c98a3bb8ab6e88be381

### PR DESCRIPTION
**Pull request by mowangdk**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Cleanup service controller unittest

#### Which issue(s) this PR fixes:

Fixes #116260

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mowangdk** [comment]
@thockin PTAL

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 1]
```diff
@@ -83,6 +83,49 @@ func defaultExternalService() *v1.Service {
 	return newService("external-balancer", types.UID("123"), v1.ServiceTypeLoadBalancer)
 }
 
+func addIngressIP(s *v1.Service) *v1.Service {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:86]: I know you're following the established pattern here, but I think this inside-out style negatively impacts the readability.  As an example, from below:

```
addServicePort(newService("udp-service", "", v1.ServiceTypeLoadBalancer), v1.ProtocolUDP),
```

vs
```
newService("udp-service", "", v1.ServiceTypeLoadBalancer, addPort(v1.ProtocolUDP)),
```

I would suggest something like:

First commit which changes this "inside-out" into the "tweaks" model, but doesn't change any coverage.
Second commit adds some tweak functions and changes callers to use them.

I would also suggest breaking this up into a series of smaller commits, each of which does one cleanup.  This is much easier to review.

Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:86]: No problem, I will modify a version as suggested

**Comment by mowangdk** [comment]
@thockin Modified as suggested, please review again. I will add the changes you submitted in [3d63aed2](https://github.com/kubernetes/kubernetes/commit/3d63aed2dca23bfec17c174b32b63c0be511fe85) after reviewing this part.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 2]
```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
...
+	}
+}
+
+func tweakAddLBI(ip string) serviceTweak {
+	return func(s *v1.Service) {
+		s.Status.LoadBalancer.Ingress = []v1.LoadBalancerIngress{{IP: ip}}
+	}
+}
+
+func tweakAddPorts(protocol v1.Protocol, targetPort string) serviceTweak {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:89]: We have a lot of this for Service but it's all implemented against v1.Service.  You can copy some logic from there.

pkg/api/service/testing/make.go

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 3]
```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
...
+	if targetPort != "" {
+		return func(s *v1.Service) {
+			s.Spec.Ports = []v1.ServicePort{{Port: 80, Protocol: protocol, TargetPort: intstr.Parse(targetPort)}}
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:92]: Why are we parsing a string to extract an integer, when we have `FromInt`.

Edit - I see it's already that way, but it's brain-damaged - can you just take an int, and use `intstr.FromInt` ?

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 4]
```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
...
+		}
+	}
+	return func(s *v1.Service) {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:95]: This whole function could be

```
sp := v1.ServicePort{Port: 80, Protocol: protocol}
if targetPort > 0 {
    sp.TargetPort = intstr.FromInt(targetPort)
}
return []v1.ServicePort{sp}
```

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 5]
```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
...
+		s.Spec.Ports = []v1.ServicePort{{Port: 80, Protocol: protocol}}
+	}
+}
+
+func tweakAddLBC(loadBalancerClass *string) serviceTweak {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:100]: LBI and LBC are not common terms.  LBIngress is OK. LBClass is OK.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 6]
```diff
@@ -145,160 +178,63 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-				Status: v1.ServiceStatus{
-					LoadBalancer: v1.LoadBalancerStatus{
-						Ingress: []v1.LoadBalancerIngress{
-							{IP: "8.8.8.8"},
-						},
-					},
-				},
-			},
+			desc:                "service no longer wants LB",
+			service:             newService("no-external-balancer", "", v1.ServiceTypeClusterIP, tweakAddLBI("8.8.8.8")),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:188]: For later cleanup -= I think this is testing an impossible case.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 7]
```diff
@@ -145,160 +178,63 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolSCTP,
-					}},
-					Type:              v1.ServiceTypeLoadBalancer,
-					LoadBalancerClass: nil,
-				},
-			},
+			desc:                 "service doesn't specify loadBalancerClass",
+			service:              newService("with-external-balancer", "", v1.ServiceTypeLoadBalancer, tweakAddLBC(nil), tweakAddPorts(v1.ProtocolSCTP, "")),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:228]: class being nil is the default, so this is identical to the case "sctp service that wants LB" above?

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 8]
```diff
@@ -66,19 +68,50 @@ func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Serv
...
+	}
+}
+
+func tweakAddLBC(loadBalancerClass *string) serviceTweak {
+	return func(s *v1.Service) {
+		s.Spec.LoadBalancerClass = loadBalancerClass
+	}
+}
+
+func tweakAddFinalizers(finalizers []string) serviceTweak {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:106]: I would write this as 

```
func tweakAddFinalizers(finalizers ...string) serviceTweak {
```

That simplifies the caller to not declare the array.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 9]
```diff
@@ -307,72 +243,30 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "service without finalizer that wants LB",
+			service:              newService("basic-service1", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, "")),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:261]: Isn't this the same as a case above?

**Review by thockin — COMMENTED** [review]
Obviously redundant tests can be removed, but do that in a new commit - either on this PR or as followup .  Your choice.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 7 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:228]: IMHO, they are both the same. so I remove this case

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 9 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:261]: yeah, it's identical with above "tcp service that wants LB"

**Comment by mowangdk** [comment]
@thockin The suggestion has been revised, please review it again.  redundant tests will be removed in next commit.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 10]
```diff
@@ -145,160 +181,63 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolSCTP,
-					}},
-					Type:              v1.ServiceTypeLoadBalancer,
-					LoadBalancerClass: nil,
-				},
-			},
+			desc:                 "service doesn't specify loadBalancerClass",
+			service:              newService("with-external-balancer", "", v1.ServiceTypeLoadBalancer, tweakAddLBClass(nil), tweakAddPorts(v1.ProtocolSCTP, 0)),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:231]: This is redundant with above

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 11]
```diff
@@ -145,160 +181,63 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-				Status: v1.ServiceStatus{
-					LoadBalancer: v1.LoadBalancerStatus{
-						Ingress: []v1.LoadBalancerIngress{
-							{IP: "8.8.8.8"},
-						},
-					},
-				},
-			},
+			desc:                 "service with finalizer that no longer wants LB",
+			service:              newService("no-external-balancer", "", v1.ServiceTypeClusterIP, tweakAddLBIngress("8.8.8.8"), tweakAddFinalizers(servicehelper.LoadBalancerCleanupFinalizer)),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:240]: maybe linewrap and put each tweak on a new line when there are more than one?

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 12]
```diff
@@ -55,8 +55,10 @@ import (
 
 const region = "us-central"
 
-func newService(name string, uid types.UID, serviceType v1.ServiceType) *v1.Service {
-	return &v1.Service{
+type serviceTweak func(s *v1.Service)
+
+func newService(name string, uid types.UID, serviceType v1.ServiceType, tweaks ...serviceTweak) *v1.Service {
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:60]: I don't think the UID argument is ever useful and just makes tests look noisy - can we get rid of it?  Just set it to a default value in here.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 13]
```diff
@@ -307,72 +246,30 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "service without finalizer that wants LB",
+			service:              newService("basic-service1", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 0)),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:264]: redundant

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 14]
```diff
@@ -307,72 +246,30 @@ func TestSyncLoadBalancerIfNeeded(t *testing.T) {
...
-				Spec: v1.ServiceSpec{
-					Ports: []v1.ServicePort{{
-						Port:     80,
-						Protocol: v1.ProtocolTCP,
-					}},
-					Type: v1.ServiceTypeLoadBalancer,
-				},
-			},
+			desc:                 "service with finalizer that wants LB",
+			service:              newService("basic-service1", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 0), tweakAddFinalizers(servicehelper.LoadBalancerCleanupFinalizer)),
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:272]: linewrap

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 15]
```diff
@@ -1567,41 +1438,15 @@ func TestNeedsUpdate(t *testing.T) {
...
-						Ports: []v1.ServicePort{{
-							Port:       80,
-							Protocol:   v1.ProtocolTCP,
-							TargetPort: intstr.Parse("22"),
-						}},
-						Type: v1.ServiceTypeLoadBalancer,
-					},
-				}
+				oldSvc = newService("tcp-service", "", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
 				newSvc = oldSvc.DeepCopy()
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:1450]: This pattern of DeepCopy and change could also be cleaned up (different commits or PRs).  You could do:

```
newSvc = copyService(oldSvc, tweakAppProtocol("http"))
```

or

```
newSvc = tweakAppProtocol("http")(oldSvd.DeepCopy())
```

but I like the first one better.

**Review by thockin — COMMENTED** [review]
This looks great - some small comments.

Let's not make this too much bigger (or if you do, make it a series of small commits, each doing ONE transformation),

As followup or as a final commit, all on its own you can do whitespace cleanups, so things like:

```
	successCases := []testCase{
		{
			// stuff
		},
		{
			// more stuff
		},
		{
			/// and so on
		},
	}
```

become:

```
	successCases := []testCase{{
		// stuff
	}, {
		// more stuff
	}, {
		/// and so on
	}}
```

10K lines of tests will become 1K or less. It's like magic.

**Comment by mowangdk** [comment]
> This looks great - some small comments.
> 
> Let's not make this too much bigger (or if you do, make it a series of small commits, each doing ONE transformation),
> 
> As followup or as a final commit, all on its own you can do whitespace cleanups, so things like:
> 
> ```
> 	successCases := []testCase{
> 		{
> 			// stuff
> 		},
> 		{
> 			// more stuff
> 		},
> 		{
> 			/// and so on
> 		},
> 	}
> ```
> 
> become:
> 
> ```
> 	successCases := []testCase{{
> 		// stuff
> 	}, {
> 		// more stuff
> 	}, {
> 		/// and so on
> 	}}
> ```
> 
> 10K lines of tests will become 1K or less. It's like magic.


sure, If I understand correctly then this pr will end up with following commits

1. change the original test to tweak mode
2. add a new service tweak method
3. add the changes related to node tweaks in #3d63aed2
4. remove duplicate tests
5. whitespace cleanups

Maybe 2 & 3 could be merged?

**Comment by thockin** [comment]
> sure, If I understand correctly then this pr will end up with following commits
> 
> 1. change the original test to tweak mode
> 2. add a new service tweak method
> 3. add the changes related to node tweaks in #3d63aed2
> 4. remove duplicate tests
> 5. whitespace cleanups
>
> Maybe 2 & 3 could be merged?

Yes those can merge.  I'm not really asking you to go back and do the following, because it's tedious, but the ideal form of this PR would be:

1. Add tweaks to `newService()`, add `tweakAddETP()`, get rid of `newETPLocalService()`, replace all callers of `newETPLocalService` with the tweak.
2. Add `tweakAddLBIngress()`, fixup all possible cases to use it
3. Add `makeServicePort` and `tweakAddPorts`, fuxup all possible cases to use it
4. repeat for `tweakAddLBClass`, `tweakAddFinalizers`, `tweakAddDeletionTimestamp`, ...
5. remove duplicate tests
6. whitespace cleanups

You would end up with something like 8 commits.  So for any FUTURE work, I think it's worthwhile to think about how to get the smallest, easiest to verify commits.

**Comment by thockin** [comment]
Here's another PR done the same way:

https://github.com/kubernetes/kubernetes/pull/116722

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 12 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:60]: sure

**Comment by mowangdk** [comment]
@thockin modified as suggested, please take a look at it again

**Comment by mowangdk** [comment]
Hi thockin, I noticed that version 1.27 has been released and codefreeze should be over, is there anything else that needs to be changed in this pr? @thockin

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 16]
```diff
@@ -1607,21 +1521,8 @@ func TestNeedsUpdate(t *testing.T) {
...
-							Port:        80,
-							Protocol:    v1.ProtocolTCP,
-							TargetPort:  intstr.Parse("22"),
-							AppProtocol: &protocol,
-						}},
-						Type: v1.ServiceTypeLoadBalancer,
-					},
-				}
+				oldSvc = newService("tcp-service", v1.ServiceTypeLoadBalancer, tweakAddPorts(v1.ProtocolTCP, 22))
+				oldSvc.Spec.Ports[0].AppProtocol = &protocol
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go]: This should also be a tweak

**Review by thockin — COMMENTED** [review]
Thanks!  It's like a fresh breeze, clearing out the old stink :)

/lgtm
/approve

**Comment by pacoxu** [comment]
/ok-to-test
/lgtm cancel 
/release-note-none

can you fix the test failure? BTW, some commits can be squashed. (Or you can add label tide/merge-method-squash  to auto-squash)

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/116277/pull-kubernetes-unit/1649513557387120640
```


1/39240 Tests Failed. | expand_less
-- | --
k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service: TestSyncLoadBalancerIfNeeded expand_less0s{Failed  === RUN   TestSyncLoadBalancerIfNeeded --- FAIL: TestSyncLoadBalancerIfNeeded (0.00s) panic: runtime error: invalid memory address or nil pointer dereference [recovered] 	panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x223b06c]  goroutine 23 [running]: testing.tRunner.func1.2({0x2434580, 0x3862690}) 	/usr/local/go/src/testing/testing.go:1526 +0x372 testing.tRunner.func1() 	/usr/local/go/src/testing/testing.go:1529 +0x650 panic({0x2434580, 0x3862690}) 	/usr/local/go/src/runtime/panic.go:890 +0x263 k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service.TestSyncLoadBalancerIfNeeded.func7(0xc0000e1900) 	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service/controller_test.go:116 +0x6c | k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service: TestSyncLoadBalancerIfNeeded expand_less | 0s | {Failed  === RUN   TestSyncLoadBalancerIfNeeded --- FAIL: TestSyncLoadBalancerIfNeeded (0.00s) panic: runtime error: invalid memory address or nil pointer dereference [recovered] 	panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x223b06c]  goroutine 23 [running]: testing.tRunner.func1.2({0x2434580, 0x3862690}) 	/usr/local/go/src/testing/testing.go:1526 +0x372 testing.tRunner.func1() 	/usr/local/go/src/testing/testing.go:1529 +0x650 panic({0x2434580, 0x3862690}) 	/usr/local/go/src/runtime/panic.go:890 +0x263 k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service.TestSyncLoadBalancerIfNeeded.func7(0xc0000e1900) 	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service/controller_test.go:116 +0x6c
k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service: TestSyncLoadBalancerIfNeeded expand_less | 0s
{Failed  === RUN   TestSyncLoadBalancerIfNeeded --- FAIL: TestSyncLoadBalancerIfNeeded (0.00s) panic: runtime error: invalid memory address or nil pointer dereference [recovered] 	panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x223b06c]  goroutine 23 [running]: testing.tRunner.func1.2({0x2434580, 0x3862690}) 	/usr/local/go/src/testing/testing.go:1526 +0x372 testing.tRunner.func1() 	/usr/local/go/src/testing/testing.go:1529 +0x650 panic({0x2434580, 0x3862690}) 	/usr/local/go/src/runtime/panic.go:890 +0x263 k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service.TestSyncLoadBalancerIfNeeded.func7(0xc0000e1900) 	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/cloud-provider/controllers/service/controller_test.go:116 +0x6c


```

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 16 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go]: okay, I'll fix the unit test, and change here by the way

**Comment by mowangdk** [comment]
> can you fix the test failure? BTW, some commits can be squashed. (Or you can add label tide/merge-method-squash to auto-squash)

okay

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 17]
```diff
@@ -113,7 +113,7 @@ func tweakAddFinalizers(finalizers ...string) serviceTweak {
 
 func tweakAddDeletionTimestamp(time time.Time) serviceTweak {
 	return func(s *v1.Service) {
-		s.ObjectMeta.DeletionTimestamp.Time = time
+		s.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: time,}
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:116]: This should be back-merged into the commit that added it

**Comment by thockin** [comment]
Let's keep the commits - they are nice and small and cost nothing.

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go (hunk 17 above)
Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:116]: ok, I tried to merge before, but found that "cleanup whitespaces" commit patch could not be merged directly after change, so I just added a new commit for simplicity.

**Comment by thockin** [comment]
Failing gofmt:

```
--- ./staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go.orig
+++ ./staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go
@@ -113,7 +113,7 @@
 
 func tweakAddDeletionTimestamp(time time.Time) serviceTweak {
 	return func(s *v1.Service) {
-		s.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: time,}
+		s.ObjectMeta.DeletionTimestamp = &metav1.Time{Time: time}
 	}
 }
```

File: staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go [hunk 18]
```diff
@@ -1,4 +1,4 @@
-/*
+0002-Chore-remove-impossible-and-redundant-cases.patch/*
```
Comment (thockin) [inline staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:1]: what's going on here?

Comment (mowangdk) [reply staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go:1]: 😵‍💫sorry,  copy mistakes, confused in the middle of the night

**Comment by mowangdk** [comment]
> /lgtm

thx~

**Pull request merged**